### ISSUE TITLE
Small enhancements

### ISSUE THREAD

**Issue by pteetor** [thomasp85/reqres#6]
@thomasp85,

I have some small enhancements for `reqres`. They were useful for incorporating `reqres` into `prairie`. I'd be grateful if you would accept them into the code base.

* Overrides for `Response$new()` defaults - Let's the caller override the defaults for status, type, and body. Streamlines construction of Response objects.
* New `as.Response` method - Lets the programmer construct a Response from other objects.

Let me know if you're interested. If so, I can rebase the feature branch and let you read the code for possible merge into master. Thanks!

**Comment by thomasp85** [comment]
I got nothing against either suggestions - I've been considering 2) for some time as I want to decouple it more from Rook... Will be happy for a PR and we can continue discussion there

**Comment by thomasp85** [comment]
Any update on this?

**Comment by pteetor** [comment]
I finished my feature branch and submitted a pull request to you (https://github.com/thomasp85/reqres/pull/7).

I am very sorry this took so long. It is very little code. This project got smothered under too many other projects. I should have cleaned this up and submitted it weeks ago.

The good news is that this code has been running for weeks on my web server, so I'm pretty sure it's solid and stable.

I created the as.Response() methods because I found them very useful for constructing real applications. Inside application, I overload as.Response() to encode application-level objects as HTTP responses, in the style of a RESTful architecture. Your mileage may very.
